plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/AwsToolkitExplorerFactory.kt

```diff
@@ -100,7 +105,20 @@ class AwsToolkitExplorerFactory : ToolWindowFactory, DumbAware {
             AwsConnectionManager.CONNECTION_SETTINGS_STATE_CHANGED,
             object : ConnectionSettingsStateChangeNotifier {
                 override fun settingsStateChanged(newState: ConnectionState) {
-                    settingsStateChanged(project, newState, toolWindow)
+                    settingsStateChanged(project, newState)
+                }
+            }
+        )
+
+        ApplicationManager.getApplication().messageBus.connect().subscribe(
+            BearerTokenProviderListener.TOPIC,
+            object : BearerTokenProviderListener {
+                override fun onChange(providerId: String, newScopes: List<String>?) {
+                    if (ToolkitConnectionManager.getInstance(project)
+                            .connectionStateForFeature(CodeCatalystConnection.getInstance()) == BearerTokenAuthState.AUTHORIZED
+                    ) {
+                        showExplorerTree(project)
+                    }
```

Comment:
same, stale project should not live beyond project scope

Reply:
add `parentDisposable=project`?

Reply:
project should not be chosen as a parent disposable
https://plugins.jetbrains.com/docs/intellij/disposers.html#choosing-a-disposable-parent

Reply:
gotcha

Reply:
didn't get a chance to look into disposable yet